File: zMore/test_conection.py
import pika

# Estabelece a conexão com o RabbitMQ
connection = pika.BlockingConnection(pika.ConnectionParameters('localhost'))
channel = connection.channel()

# Cria uma fila (se não existir)
channel.queue_declare(queue='test_queue', durable=True)

# Publica uma mensagem na fila
channel.basic_publish(
    exchange='',
    routing_key='test_queue',
    body='Hello RabbitMQ!',
    properties=pika.BasicProperties(
        delivery_mode=2,  # Torna a mensagem persistente
    )
)

print("Mensagem enviada!")

# Fecha a conexão
connection.close()

File: zMore/test_conection.ts
import amqp from 'amqplib';

async function createQueueAndSendMessage() {
  try {
    // Estabelece a conexão com o RabbitMQ
    const connection = await amqp.connect('amqp://localhost');
    const channel = await connection.createChannel();

    const queue = 'test_queue';
    const message = 'Hello RabbitMQ from TypeScript!';

    // Declara a fila (se não existir)
    await channel.assertQueue(queue, {
      durable: true, // Fila será persistente
    });

    // Envia a mensagem para a fila
    channel.sendToQueue(queue, Buffer.from(message), {
      persistent: true, // A mensagem será persistente
    });

    console.log(`Mensagem enviada para a fila ${queue}: ${message}`);

    // Fecha a conexão após um pequeno delay
    setTimeout(() => {
      channel.close();
      connection.close();
    }, 500);
  } catch (error) {
    console.error('Erro ao conectar ou enviar mensagem:', error);
  }
}

// Chama a função
createQueueAndSendMessage();
